src/client/mod.rs

```diff
@@ -74,9 +74,8 @@ impl StateStream for ResponseStream {
     type State = Client;
     type Error = io::Error;
     fn poll(&mut self) -> Poll<StreamEvent<Self::Item, Self::State>, Self::Error> {
-        if self.future.is_some() {
-            let mut future = self.future.take().unwrap();
-            match future.poll() {
+        match self.future.take() {
+            Some(mut future) => match future.poll() {
```

Comment:
`self.future.map(|future| { ... })` would be even more concise, wouldn’t it?

Reply:
Maybe -- it's not a great fit conceptually, though, since fundamentally we're taking out the future here rather than converting it to an `Option` of a different type.